appengine/go11x/pubsub/authenicated_push/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"google.golang.org/api/idtoken"
+	"google.golang.org/api/option"
+)
+
+var (
+	// Messages received by this instance.
+	messagesMu sync.Mutex
+	messages   [][]byte
+
+	// defaultHTTPClient aliases http.DefaultClient for testing
+	defaultHTTPClient = http.DefaultClient
+)
+
+const maxMessages = 10
+
+func main() {
+	http.HandleFunc("/pubsub/message/list", listHandler)
+	http.HandleFunc("/pubsub/message/receive", receiveMessagesHandler)
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	if err := http.ListenAndServe(":"+port, nil); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// pushRequest represents the payload of a Pub/Sub push message.
+type pushRequest struct {
+	Message      message `json:"message"`
+	Subscription string  `json:"subscription"`
+}
+
+type message struct {
+	Attributes map[string]string `json:"attributes"`
+	Data       []byte            `json:"data"`
+	ID         string            `json:"messageId"`
+}
+
+// [START gae_standard_pubsub_auth_push]
+// receiveMessagesHandler validates authentication token and caches the Pub/Sub
+// message received.
+func receiveMessagesHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+		return
+	}
+
+	pubsubVerificationToken := os.Getenv("PUBSUB_VERIFICATION_TOKEN")
```

Comment:
Perhaps check this is set during app startup?

Reply:
Sure.